odl/solvers/functional/functional.py

```diff
@@ -0,0 +1,1471 @@
+                super().__init__(functional.domain, functional.domain,
+                                 linear=False)
+
+            def _call(self, x):
+                """Evaluates the gradient in a point ``x``.
+
+                Parameters
+                ----------
+                x : `LinearSpaceVector`
+                    Element in the domain of the operator.
+
+                Returns
+                -------
+                out : `LinearSpaceVector`
+                    The gradient in the point ``x``. An element in the doimain
+                    of the operator.
+                """
+                return functional._scalar * functional._func.gradient(x)
+
+        return LeftScalarMultGradient()
+
+    @property
+    def conjugate_functional(self):
+        """Convex conjugate functional of the scaled functional."""
+        # The helper function only allows positive scaling parameters.
+        # Otherwise it gives an error.
+        return ConvexConjugateFuncScaling(
+            self._func.conjugate_functional, self._scalar)
+
+    def proximal(self, sigma=1.0):
+        """Return the proximal operator of the scaled functional.
+
+        Parameters
+        ----------
+        sigma : positive float, optional
+            Regularization parameter of the proximal operator.
+
+        Returns
+        -------
+        out : `Operator`
+            Domain and range equal to domain of functional.
+        """
+        sigma = float(sigma)
+        if sigma * self._scalar < 0:
+            raise ValueError('The step lengt {} times the scalar {} needs to '
+                             'be nonnegative.'.format(sigma, self._scalar))
+        return self._func.proximal(sigma * self._scalar)
+
+    def derivative(self, point):
+        """Returns the derivative operator in the given point.
+
+        This function returns the linear operator
+
+            ``x --> <x, grad_f(point)>``,
+
+        where ``grad_f(point)`` is the gradient of the functional in the point
+        ``point``.
+
+        Parameters
+        ----------
+        point : `LinearSpaceVector`
+            The point in which the gradient is evaluated.
+
+        Returns
+        -------
+        out : `DerivativeOperator`
+            The linear operator that maps ``x --> <x, grad_f(point)>``.
+        """
+        return self._scalar * self._func.derivative(point)
+
+
+class FunctionalRightScalarMult(Functional, OperatorRightScalarMult):
+    """Scalar multiplication of the argument of functional.
+
+    Given a functional ``f`` and a scalar ``scalar``, this represents the
+    functional
+
+        ``(f * scalar)(x) = f(scalar * x)``.
+    """
+
+    def __init__(self, func, scalar):
+        """Initialize a FunctionalRightScalarMult instance.
+
+        Parameters
+        ----------
+        scal : `Scalar`
+            Scalar argument
+        func : `Functional`
+            The left ("outer") functional
```

Comment:
Change order in doc parameter list to match call signature. Also update the doc for `func`.